### PR TITLE
feat(cli): adding Thor CLI with SOPS

### PR DESCRIPTION
**Pull request by codefriar**
Deployment is going to be from within a docker container (only dependency docker)
The ansible-vault is removed and replaced by SOPS (https://github.com/mozilla/sops) which removes the need of the vault.yml and encrypts only the values of the variables

BREAKING CHANGE: Task-it CLI (bash based) changes to Thor Cli (ruby based)

### CODE REVIEW

File: roles/massivedecks/templates/docker-compose.massivedecks.yml.j2 [hunk 1]
```diff
@@ -11,12 +11,8 @@ networks:
 
 services:
   server:
-<<<<<<< HEAD
     container_name: massivedecks_server
-    image: massivedecks/server:{{ massivedecks.server_version}}
```
Comment (voryzen) [inline roles/massivedecks/templates/docker-compose.massivedecks.yml.j2:16]: Nice catch

File: tasks/FrictionLog.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Friction Log
```
Comment (voryzen) [inline tasks/FrictionLog.md:1]: Are we keeping the friction log in here?

**Comment by denis-ev** [comment]
We should include the changes of the docs, so we don't forget it.
If so, should be merge the docs change first? After you guys checked it out and we changed or removed the banner image.
What do you guys think?

File: launch.sh [hunk 3]
```diff
@@ -18,4 +18,4 @@ docker run --rm -it \
   -v "$HOME/.ssh/$(pwless_sshkey).pub":"/root/.ssh/$(pwless_sshkey).pub" \
   -v $(pwd):/data \
   -v $HOME/.vlab_vault_pass:/ansible_vault_pass \
-  ef410c4e56a3 /bin/bash
+  36281de0f837 /bin/bash
```
Comment (denis-ev) [inline launch.sh:21]: This looks like the dynamic container name, this should have a name instead.

**Comment by denis-ev** [comment]
can we change the Classes to all small letters?
To run the commands we only have to type `vlab service ...` and not `vlab Service ...`.
I personally don't like capital letters in cli commands. haha

**Comment by denis-ev** [comment]
I just rebased the whole thing.

**Comment by voryzen** [comment]
Just in case there are any mistakes.. (I just almost did it) I'm turning auto merge off.

We can always turn it back on when the WIP tag/label is removed

**Comment by denis-ev** [comment]
Haha no worries.

**Comment by voryzen** [comment]
> Haha no worries.

haha, I thought that I had destroyed the whole thing.

I was very sad

File: tasks/FrictionLog.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (denis-ev) [reply tasks/FrictionLog.md:1]: @codefriar

**Comment by Kitsu01** [comment]
I am not sure if I should open a bug or just write here so I will choose the simple path.
When I clone the repo, checkout this branch and launch ./launcher.sh I get dropped inside a docker container with bash and nothing happens. When simply running ./vlab, I get:
```
[VivumLab]$ ./vlab
Traceback (most recent call last):
	2: from ./vlab:6:in `<main>'
	1: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- bundler (LoadError)
```

This happens also with sudo and ruby is installed.

Did I miss something?

**Comment by codefriar** [comment]
@Kitsu01 

it depends on what you're trying to do, but I suspect you've identified an edge case. In general, *you* shouldn't need ruby, etc. Because all of that is provided for and setup in the docker image your launch. 

However, we're in the midst of the transition to:
- a new docker image
- a new cli
- a new way of thinking about the CLI and deploying.

As such, if you just run ./launch.sh the environment isn't what you expect or need.

*until further notice* if you're going to test the new cli with us, you need to execute *one* of the following commands.

## 1st time:

```./launch.sh local```

## Subsequent times

``` ./launch.sh local start ```

**Review by denis-ev — DISMISSED** [review]

**Review by denis-ev — DISMISSED** [review]

**Review by denis-ev — APPROVED** [review]

**Review by denis-ev — DISMISSED** [review]

**Review by denis-ev — DISMISSED** [review]

**Review by denis-ev — APPROVED** [review]

**Review by denis-ev — DISMISSED** [review]

**Review by denis-ev — APPROVED** [review]

**Review by denis-ev — APPROVED** [review]

**Review by denis-ev — DISMISSED** [review]

**Review by voryzen — APPROVED** [review]

**Pull request merged**